integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/limitations/ConcurrentEmbeddedUpdateLimitationIT.java

```diff
@@ -61,6 +63,9 @@ public void indexingStrategySession() throws Throwable {
 				.withProperty( HibernateOrmMapperSettings.AUTOMATIC_INDEXING_SYNCHRONIZATION_STRATEGY, "sync" )
 				.setup( Book.class, Author.class, BookEdition.class );
 
+		HibernateORMSkipTestHelper.skipForDialect( sessionFactory, SQLServerDialect.class,
```

Comment:
You could perform this check before creating the session factory, which would avoid a bunch of connections and schema creation.

That's what we did for multi-tenancy tests, cf. `MultitenancyTestHelper`.

Just add this to `org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmSetupHelper.SetupContext`:

```java
		public SetupContext skipTestForDialect(Class<? extends Dialect> dialect, String reason) {
			withConfiguration( b -> b.onMetadata( metadataImplementor -> {
				Dialect currentDialect = metadataImplementor.getDatabase().getDialect();
				assumeFalse( "Skipping test for dialect " + dialect.getName() + "; reason: " + reason, dialect.isAssignableFrom( currentDialect.getClass() ) );
			} ) );
			return thisAsC();
		}
```

And call it before `setup( Book.class, ... )` just above.